server/src/main/java/org/candlepin/resource/OwnerResource.java

```diff
@@ -1735,23 +1736,34 @@ public SystemPurposeAttributesDTO getSyspurpose(
         @Verify(Owner.class) String ownerKey) {
         Owner owner = findOwnerByKey(ownerKey);
-        List<String> consumerRoles = this.consumerCurator.getDistinctSyspurposeRolesByOwner(owner);
-        List<String> consumerUsages = this.consumerCurator.getDistinctSyspurposeUsageByOwner(owner);
-        List<String> consumerSLAs = this.consumerCurator.getDistinctSyspurposeServicelevelByOwner(owner);
-        List<String> consumerAddons = this.consumerCurator.getDistinctSyspurposeAddonsByOwner(owner);
-
-        Map<String, Set<String>> dtoMap = new HashMap<>();
-        Arrays.stream(SystemPurposeAttributeType.values())
-            .forEach(x -> dtoMap.put(x.toString(), new LinkedHashSet<>()));
-
-        dtoMap.get(SystemPurposeAttributeType.ROLES.toString()).addAll(consumerRoles);
-        dtoMap.get(SystemPurposeAttributeType.USAGE.toString()).addAll(consumerUsages);
-        dtoMap.get(SystemPurposeAttributeType.SERVICE_LEVEL.toString()).addAll(consumerSLAs);
-        dtoMap.get(SystemPurposeAttributeType.ADDONS.toString()).addAll(consumerAddons);
-
         SystemPurposeAttributesDTO dto = new SystemPurposeAttributesDTO();
-        dto.setOwner(translator.translate(owner, NestedOwnerDTO.class));
-        dto.setSystemPurposeAttributes(dtoMap);
+        try {
+            List<String> consumerRoles = this.consumerCurator.getDistinctSyspurposeValuesByOwner(owner,
+                Consumer_.ROLE);
+            List<String> consumerUsages = this.consumerCurator.getDistinctSyspurposeValuesByOwner(owner,
+                Consumer_.USAGE);
+            List<String> consumerSLAs = this.consumerCurator.getDistinctSyspurposeValuesByOwner(owner,
+                Consumer_.SERVICE_LEVEL);
+            List<String> consumerServiceTypes = this.consumerCurator.getDistinctSyspurposeValuesByOwner(owner,
+                Consumer_.SERVICE_TYPE);
+            List<String> consumerAddons = this.consumerCurator.getDistinctSyspurposeAddonsByOwner(owner);
+
+            Map<String, Set<String>> dtoMap = new HashMap<>();
+            Arrays.stream(SystemPurposeAttributeType.values())
+                .forEach(x -> dtoMap.put(x.toString(), new LinkedHashSet<>()));
+
+            dtoMap.get(SystemPurposeAttributeType.ROLES.toString()).addAll(consumerRoles);
+            dtoMap.get(SystemPurposeAttributeType.USAGE.toString()).addAll(consumerUsages);
+            dtoMap.get(SystemPurposeAttributeType.SERVICE_LEVEL.toString()).addAll(consumerSLAs);
+            dtoMap.get(SystemPurposeAttributeType.ADDONS.toString()).addAll(consumerAddons);
+            dtoMap.get(SystemPurposeAttributeType.SERVICE_TYPE.toString()).addAll(consumerServiceTypes);
+
+            dto.setOwner(translator.translate(owner, NestedOwnerDTO.class));
+            dto.setSystemPurposeAttributes(dtoMap);
+        }
+        catch (NotFoundException e) {
+            throw new NotFoundException("Unrecognized system purpose attribute.");
```

Comment:
We shouldn't be throwing such an error on the API, because the user never specified any syspurpose attribute; we, the developer, hardcoded the list of attributes we want to gather. We should probably remove this block from here, and have consumerCurator.getDistinctSyspurposeValuesByOwner() throw a RuntimeException, which doesn't have to be checked here.

Reply:
Updated.